tests/erica_legacy/pyeric/test_eric_errors.py

```diff
@@ -276,3 +277,22 @@ def test_if_empty_xml_then_raise_no_error(self):
             check_xml(b'')
         except EricProcessNotSuccessful:
             self.fail("CheckXml raised EricProcessNotSuccessful unexpected.")
+
+
+class TestGetErrorCodesFromServerErrMsg:
+
+    def test_if_no_error_code_in_server_err_msg_then_return_empty_array(self):
+        server_err_msg = '<?xml version="1.0" encoding="UTF-8"?><EricGetErrormessagesFromXMLAnswer xmlns="http://www.elster.de/EricXML/1.0/EricGetErrormessagesFromXMLAnswer"></EricGetErrormessagesFromXMLAnswer>'
+        result = get_error_codes_from_server_err_msg(server_err_msg)
+        assert result == []
+
+    def test_if_one_error_code_in_server_err_msg_then_return_code_in_array(self):
+        server_err_msg = '<?xml version="1.0" encoding="UTF-8"?><EricGetErrormessagesFromXMLAnswer xmlns="http://www.elster.de/EricXML/1.0/EricGetErrormessagesFromXMLAnswer">\t<Fehler>\t\t<Code>01</Code>\t\t<Meldung>Ein Fehler ist aufgetreten</Meldung>\t</Fehler>   </EricGetErrormessagesFromXMLAnswer>'
+        result = get_error_codes_from_server_err_msg(server_err_msg)
+        assert result == ['01']
+
+    def test_if_many_error_code_in_server_err_msg_then_return_all_codes_in_array(self):
+        server_err_msg = '<?xml version="1.0" encoding="UTF-8"?><EricGetErrormessagesFromXMLAnswer xmlns="http://www.elster.de/EricXML/1.0/EricGetErrormessagesFromXMLAnswer">\t<Fehler>\t\t<Code>01</Code>\t\t<Meldung>Ein Fehler ist aufgetreten</Meldung>\t</Fehler>   <Fehler>\t\t<Code>02</Code>\t\t<Meldung>Ein weiterer Fehler ist aufgetreten</Meldung>\t</Fehler></EricGetErrormessagesFromXMLAnswer>'
```

Comment:
Are those real Elster XMLs that are possibly returned for a transfer error? I think they're rather self-made, no? I feel like it would be safer to actually have a possible response in our /samples folder and actually pass that to the function - maybe just as an additional test.

Reply:
This is a sample xml. I just changed the error codes to make it more readable

Reply:
I see. Great that this is already merged then. 👍  :D